Mathlib/Topology/ContinuousFunction/ContinuousMapZero.lean

```diff
@@ -34,12 +36,7 @@ variable [TopologicalSpace X] [TopologicalSpace Y] [TopologicalSpace R]
 
 instance instFunLike : FunLike C(X, R)₀ X R where
   coe f := f.toFun
-  coe_injective' f g h := by
-    cases f
-    cases g
-    congr
-    apply DFunLike.coe_injective'
-    exact h
+  coe_injective' _ _ h := congr(⟨⟨$(h), _⟩, _⟩)
```

Comment:
shouldn't there be a space before the opening parenthesis?

Reply:
(I tried, and the answer is no, but I don't understand the trick — what does $ do?)

Reply:
`congr(...)` is special syntax for creating congruence lemmas. For example, if `h : a = b`, then `congr(f $h)` gives that `f a = f b`. No space is how it's distinguished from the `congr` lemma.

The congruence in question can be written as `congr(⟨⟨$h, _⟩, _⟩)` without the additional parentheses.